core/src/main/scala/org/apache/spark/util/RpcUtils.scala

```diff
@@ -32,4 +35,24 @@ object RpcUtils {
     Utils.checkHost(driverHost, "Expected hostname")
     rpcEnv.setupEndpointRef(driverActorSystemName, RpcAddress(driverHost, driverPort), name)
   }
+
+  /** Returns the configured number of times to retry connecting */
+  def numRetries(conf: SparkConf): Int = {
+    conf.getInt("spark.rpc.num.retries", 3)
```

Comment:
we should do spark.rpc.numRetries instead (you might need to translate this to akka.num.retries if we pass this back to akka)